From dd9ec5babe85dcc0ae2f3ee8441f495c485936e6 Mon Sep 17 00:00:00 2001
From: Ian <wei0131@realtek.com>
Date: Thu, 5 Mar 2015 15:14:55 +0530
Subject: [PATCH] ASoC: rt5670: Add the PDM ctrl settings

Add the PDM control settings to disable static noise.

Issue: GMINL-6938

Change-Id: Id06a2bbf954e9b6110ad331ad00651ac38dbf38e
Signed-off-by: Ian <wei0131@realtek.com>
Signed-off-by: Samreen Nilofer <samreen.nilofer@intel.com>
---
 sound/soc/codecs/rt5670.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/sound/soc/codecs/rt5670.c b/sound/soc/codecs/rt5670.c
index cc071b6..2516871 100644
--- a/sound/soc/codecs/rt5670.c
+++ b/sound/soc/codecs/rt5670.c
@@ -87,6 +87,7 @@ static struct rt5670_init_reg init_list[] = {
 	{ RT5670_STO1_ADC_MIXER	, 0x5940 }, /* DMIC2 setting */
 	{ RT5670_STO1_ADC_DIG_VOL, 0xafaf }, /* Mute STO1 ADC for depop */
 	{ RT5670_PDM_OUT_CTRL	, 0xff01 },
+	{ RT5670_PDM_OUT_CTRL   , 0xff00 },
 #ifdef JD1_FUNC
 	{ RT5670_GPIO_CTRL2	, 0x0004 },
 	{ RT5670_GPIO_CTRL1	, 0x8000 },
@@ -2829,9 +2830,6 @@ static int get_clk_info(int sclk, int rate)
 {
 	int i, pd[] = {1, 2, 3, 4, 6, 8, 12, 16};
 
-#ifdef USE_ASRC
-	return 0;
-#endif
 	if (sclk <= 0 || rate <= 0)
 		return -EINVAL;
 
-- 
1.9.1

